### PR TITLE
Switch to micromatch

### PR DESCRIPTION
**Pull request by es128**
@jonschlinkert would appreciate a review before I publish this

### CODE REVIEW

**Comment by jonschlinkert** [comment]
Sure thing I'll be online in 30 min or so

**Comment by jonschlinkert** [comment]
sorry didn't have a chance to look earlier, will look over now.

**Comment by es128** [comment]
Thanks. There's no rush.

File: index.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-  if (!Array.isArray(criteria)) { criteria = [criteria]; }
-  if (arguments.length === 1) { return anymatch.bind(null, criteria); }
-  var string = Array.isArray(value) ? value[0] : value;
-  if (!startIndex) { startIndex = 0; }
+  criteria = arrify(criteria);
+  value = arrify(value);
+  if (arguments.length === 1) {
+    return criteria.length === 1 ?
+      micromatch.matcher(criteria[0]) :
+      anymatch.bind(null, criteria.map(function(criterion) {
```
Comment (es128) [inline index.js]: @jonschlinkert btw have you considered adding array support to `micromatch.matcher`? It would make it possible to simplify this bit.

**Comment by jonschlinkert** [comment]
Just spent some time with it, there is nothing I would change in the code it looks great. 

maybe consider doing a major bump though, even though the API isn't changing. I did a million unit tests for micromatch and tried to make sure there was parity, but just being realistic I still expect there to be some edge cases I didn't notice. 

minimatch has a number of bugs, but people are used to them. obviously your call if you're okay with getting hit with an issue or two if that were to happen... on my end I'll make sure to fix whatever comes up right away. thanks!

**Comment by es128** [comment]
I'll think a bit more on whether to bump major. 

Either way, I'm going to use latest in chokidar, so any issues will wind up affecting someone and we'll hear about them.

**Comment by jonschlinkert** [comment]
> so any issues will wind up affecting someone and we'll hear about them.

lol, well, on the upside there is no faster way for me to flesh out the bugs and improve micromatch :)

**Comment by jonschlinkert** [comment]
... fwiw I use micromatch on everything now. I have projects that have been using if for a couple of months and I haven't had any issues. I'm just conservative.

it will be interesting to see if you notice a speed increase with chokidar. Micromatch spends more time on optimizing regexes, so it takes longer to build the regex, but once it's built the matching is much faster. I would expect that this approach would have the most obvious benefit in something chokidar, since the regex is cached and doesn't have to be re-built after the first time.

**Comment by es128** [comment]
Chokidar's glob support is new, but it isn't ever the bottleneck compared with all the fs operations. So I wouldn't expect a really noticeable impact. Still, every bit counts.
